coderd/workspaceagents_test.go

```diff
@@ -1053,6 +1055,87 @@ func TestWorkspaceAgentListeningPorts(t *testing.T) {
 	})
 }
 
+func TestWorkspaceAgentContainers(t *testing.T) {
+	t.Parallel()
+
+	if runtime.GOOS != "linux" {
+		t.Skip("this test creates containers, which is flaky on non-linux runners")
+	}
+
+	pool, err := dockertest.NewPool("")
```

Comment:
I'd prefer to use a fake container here if we need it, or perhaps a fake API response from the agent, instead of introducing docker dependency to coderd tests.

Reply:
I kept the existing test but hid it behind an environment variable. Added a separate test and plumbed through a mock implementation.